src/wiki/core/wiki_paginate.py

```diff
@@ -0,0 +1,30 @@
+from django.core.paginator import Paginator
+
+class WikiPaginator(Paginator):
+
+    def __init__(self, *args, **kwargs):
+        """
+        :param side_pages: How many pages should be shown before and after the current page
+        """
+        self.side_pages = kwargs.pop('side_pages', 4)
+        super(WikiPaginator, self).__init__(*args, **kwargs)
+
+    def _get_page(self, *args, **kwargs):
+        self.curPage = super(WikiPaginator, self)._get_page(*args, **kwargs)
```

Comment:
@benjaoming Calling _get_page() from page_range() unfortunately does not work.
Source from the django paginator (django/core/paginator.py):

    class Paginator(object):
        def _get_page(self, *args, **kwargs):
            """
            Returns an instance of a single page. This hook can be used by
            subclasses to use an alternative to the standard :cls:`Page` object.
            """
            return Page(*args, **kwargs)

    class Page(collections.Sequence):
        def __init__(self, object_list, number, paginator):
            self.object_list = object_list
            self.number = number
            self.paginator = paginator

I.e. the page number is required. And that is exactly what is needed here in the
first place. Based on that I would say that it is guaranteed that _get_page()
is called before page_range().